### PR TITLE
Ct 30 check for unsaved changes in nuke before submitting

### PR DESCRIPTION
**Pull request by easythrees**

### CODE REVIEW

File: conductor/submitter_nuke.py [hunk 1]
```diff
@@ -223,6 +223,19 @@ def getOutputPath(self):
         output_path = file_utils.get_common_dirpath(write_paths)
         return output_path, write_paths
 
+    def checkSaveBeforeSubmission(self):
+        '''
+        Check if scene has unsaved changes and prompt user if they'd like to
+        save their Maya scene before continuing
```
[the remark this answers is not in the document]
Comment (easythrees) [reply conductor/submitter_nuke.py]: Crap! I'll change it now....

[the remark this answers is not in the document]
Comment (easythrees) [reply conductor/submitter_nuke.py]: Okay, new change pushed.

File: conductor/lib/nuke_utils.py [hunk 2]
```diff
@@ -166,6 +166,17 @@ def get_nuke_script_filepath():
...
+    '''
+    Check if the scene's been modified, and error out if it has been
+    '''
+    return nuke.root().modified()
+
+def save_current_nuke_script():
+    '''
+    Save the current script
+    '''
+    nuke.scriptSave()
```
Comment (flebel) [inline conductor/lib/nuke_utils.py]: Can you return the value returned from `scriptSave`? https://learn.foundry.com/nuke/developers/70/pythonreference/

File: conductor/submitter_nuke.py [hunk 3]
```diff
@@ -236,6 +249,12 @@ def runPreSubmission(self):
         again (after validation succeeds), we also pass the depenencies along
         in the returned dictionary (so that we don't need to collect them again).
         '''
+        # Check if script has unsaved changes and ask user if they'd like to
+        # save their script before continuing with submission
+        if not self.checkSaveBeforeSubmission():
+            raise exceptions.UserCanceledError()
+        else:
```
Comment (flebel) [inline conductor/submitter_nuke.py]: No need for an `else` here if the body of the sibling `if` raises an exception.

File: conductor/submitter_nuke.py [hunk 4]
```diff
@@ -223,6 +223,19 @@ def getOutputPath(self):
...
 
+    def checkSaveBeforeSubmission(self):
+        '''
+        Check if script has unsaved changes and prompt user if they'd like to
+        save it before continuing
+        '''
+        file_unsaved = nuke_utils.check_script_modified()
+        if file_unsaved:
+            title = "Unsaved Nuke Script Data"
+            message = "Save Nuke file script submitting?"
```
Comment (flebel) [inline conductor/submitter_nuke.py]: `Save Nuke file script submitting?` should probably say
`Save Nuke script before submitting?`

Comment (lawschlosser) [reply conductor/submitter_nuke.py]: ha, good catch!

File: conductor/submitter_nuke.py (hunk 3 above)
Comment (lawschlosser) [reply conductor/submitter_nuke.py]: agreed.  Ravi was simply copying the code from the maya submitter (which has the same issue)

File: conductor/submitter_nuke.py (hunk 4 above)
Comment (easythrees) [reply conductor/submitter_nuke.py]: Done

File: conductor/submitter_nuke.py (hunk 3 above)
Comment (easythrees) [reply conductor/submitter_nuke.py]: Removed

File: conductor/lib/nuke_utils.py (hunk 2 above)
Comment (easythrees) [reply conductor/lib/nuke_utils.py]: done

File: conductor/submitter_nuke.py [hunk 5]
```diff
@@ -253,8 +253,6 @@ def runPreSubmission(self):
         # save their script before continuing with submission
         if not self.checkSaveBeforeSubmission():
             raise exceptions.UserCanceledError()
-        else:
-            nuke_utils.save_current_nuke_script()
```
Comment (flebel) [inline conductor/submitter_nuke.py]: I don't think you meant to remove the `nuke_utils.save_current_nuke_script()` line, but just the `else:` block, and dedent `nuke_utils.save_current_nuke_script()`.

Comment (easythrees) [reply conductor/submitter_nuke.py]: Okay, maybe I do need a coffee this morning... Sorry about that.

File: conductor/submitter_nuke.py [hunk 6]
```diff
@@ -223,6 +223,19 @@ def getOutputPath(self):
...
 
+    def checkSaveBeforeSubmission(self):
+        '''
+        Check if script has unsaved changes and prompt user if they'd like to
+        save it before continuing
+        '''
+        file_unsaved = nuke_utils.check_script_modified()
+        if file_unsaved:
+            title = "Unsaved Nuke Script Data"
+            message = "Save Nuke file script before submitting?"
```
Comment (flebel) [inline conductor/submitter_nuke.py]: Can you change `Save Nuke file script before submitting?` to `Save Nuke script before submitting?`?

**Pull request merged**